.github/workflows/code-review.yml

```
@@ -0,0 +1,34 @@
+name: AI Code Review
+
+on:
+  pull_request:
+    types: [opened, synchronize]
+
+permissions:
+  pull-requests: write
+  contents: read
+  checks: write
+
+jobs:
+  review:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install --upgrade PyGithub google-genai requests python-dotenv
```

It's good practice to pin the exact version of the action instead of just the major version to ensure consistent behavior over time.

```suggestion
actions/setup-python@v4.7.0 (or the latest specific version)
```